src/test/java/org/springframework/guice/ScopingTests.java

```diff
@@ -0,0 +1,85 @@
+package org.springframework.guice;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.Test;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.guice.ScopingTests.CustomScope;
+import org.springframework.guice.ScopingTests.SomeCustomScopeDependency;
+import org.springframework.guice.ScopingTests.SomeNoScopeDependency;
+import org.springframework.guice.ScopingTests.SomeSingletonDependency;
+import org.springframework.guice.annotation.EnableGuiceModules;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.Provider;
+import com.google.inject.Scope;
+import com.google.inject.util.Providers;
+
+public class ScopingTests {
+
+	@Test
+	public void verifyScopes() {
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(
+				ScopingTestsConfig.class);
+		SomeSingletonDependency someSingletonDependency1 = context.getBean(SomeSingletonDependency.class);
+		SomeSingletonDependency someSingletonDependency2 = context.getBean(SomeSingletonDependency.class);
+		
+		assertNotNull(someSingletonDependency1);
+		assertNotNull(someSingletonDependency2);
+		assertEquals(someSingletonDependency1, someSingletonDependency2);
+		
+		SomeNoScopeDependency someNoScopeDependency1 = context.getBean(SomeNoScopeDependency.class);
+		SomeNoScopeDependency someNoScopeDependency2 = context.getBean(SomeNoScopeDependency.class);
+		
+		assertNotNull(someNoScopeDependency1);
+		assertNotNull(someNoScopeDependency2);
+		assertNotEquals(someNoScopeDependency1, someNoScopeDependency2);
+		
+		SomeCustomScopeDependency someCustomScopeDependency1 = context.getBean(SomeCustomScopeDependency.class);
+		SomeCustomScopeDependency someCustomScopeDependency2 = context.getBean(SomeCustomScopeDependency.class);
+		
+		assertNotNull(someCustomScopeDependency1);
+		assertNotNull(someCustomScopeDependency2);
+		assertEquals(someCustomScopeDependency1, someCustomScopeDependency2);
```

Comment:
Shouldn't this be NotEqual? I thought custom scopes should not be singletons.

Reply:
I implemented my custom scope to be a singleton. Guess I can make it be prototype.